### PR TITLE
[PLUGIN-470] Cherry pick into 0.16

### PR DESCRIPTION
**Pull request by fernst**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -73,7 +73,7 @@
     <bigquery.connector.hadoop2.version>hadoop2-1.0.0</bigquery.connector.hadoop2.version>
     <commons.codec.version>1.4</commons.codec.version>
     <cdap.version>6.3.0</cdap.version>
-    <cdap.plugin.version>2.5.0</cdap.plugin.version>
+    <cdap.plugin.version>2.5.1-SNAPSHOT</cdap.plugin.version>
```
Comment (CuriousVini) [inline pom.xml:76]: why is this change needed? usually we do not upgrade it unless there is an api change

**Review by CuriousVini — APPROVED** [review]
lgtm

File: pom.xml (hunk 1 above)
Comment (fernst) [reply pom.xml:76]: Discussed offline that this is related to the changes introduced in https://github.com/cdapio/hydrator-plugins/pull/1324
